internal/dbutil/sqliteutil/query.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package sqliteutil
+
+import (
+	"database/sql"
+	"regexp"
+	"strings"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/internal/dbutil/dbschema"
+)
+
+// Queryer is a representation for something that can query.
+type Queryer interface {
+	// Query executes a query that returns rows, typically a SELECT.
+	Query(query string, args ...interface{}) (*sql.Rows, error)
+}
+
+// QuerySchema loads the schema from postgres database.
+func QuerySchema(tx Queryer) (*dbschema.Schema, error) {
+	schema := &dbschema.Schema{}
+
+	type Definition struct {
+		name string
+		sql  string
+	}
+
+	tableDefinitions := make([]*Definition, 0)
+	indexDefinitions := make([]*Definition, 0)
+
+	// find tables and indexes
+	err := func() error {
+		rows, err := tx.Query(`
+			SELECT name, type, sql FROM sqlite_master WHERE sql NOT NULL AND name NOT LIKE 'sqlite_%'
+		`)
+		if err != nil {
+			return err
+		}
+		defer func() { err = errs.Combine(err, rows.Close()) }()
+
+		for rows.Next() {
+			var defName, defType, defSQL string
+			err := rows.Scan(&defName, &defType, &defSQL)
+			if err != nil {
+				return err
+			}
+			if defType == "table" {
+				tableDefinitions = append(tableDefinitions, &Definition{name: defName, sql: defSQL})
```

Comment:
You can add the partial table and index directly to the schema, that way you don't have to have the intermediate Definition.

Reply:
in schema there is no place for `sql`